lib/Dialect/OM/Transforms/LinkModules.cpp

```diff
@@ -273,6 +416,18 @@ void LinkModulesPass::runOnOperation() {
       }
   }
 
+  for (auto &[name, symOps] : symbolToOps) {
+
+    if (symOps.size() > 1)
+      for (auto *op : symOps) {
+        if (isa<HWModuleLike, ClassLike>(op))
+          continue;
+        auto enclosingModule = cast<mlir::ModuleOp>(op->getParentOp());
```

Comment:
I think you could do `auto enclosingModule = op->getParentOfType<mlir::ModuleOp>()`

Reply:
Thanks, updated.